app/mailers/job_mailer.rb

```diff
@@ -6,6 +6,16 @@ class JobMailer < ApplicationMailer
   def completion_email
     @job_run = params[:job_run]
     @user = @job_run.batch_context.user
+    @completed_accessions = completed_accessions
+    @failed_accessions = failed_accessions
     mail(to: @user.email, subject: @job_run.mail_subject)
   end
+
+  def completed_accessions
+    @job_run.accessions.select { |acc| acc[:state] == 'completed' }
```

Comment:
i think you could do this in query-land without loading all the `accessions` and using a ruby `select`, by doing something like `@job_run.accessions.where(state: 'completed')` (and i think [ActiveRecord's `enum` actually provides some scope magic](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html) that'd let you do e.g. `@job_run.accessions.completed`).

i'm not sure how many `Accessions` we expect to hang off of a `JobRun`.  probably few enough that this doesn't really matter from a performance PoV, and what's here is still readable, so def wouldn't request changes over this, just mentioning for consideration.

Reply:
actually, looking at the model/schema and the rest of the code, i'm remembering that an `Accession` is one druid, not the whole batch being accessioned by the job.  so actually it might be noticeably better to filter on the SQL query side since there could be up to a few hundred or even a few thousand associated druids?  still wouldn't block the PR over this, though.  retrieving e.g. 5k rows on occasion in an async job is unlikely to crash things.